db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/sqlgrammar.jj

```diff
@@ -9023,7 +9023,17 @@ ValueNode
 
             return value;
       }
-     | charType = charOrVarchar() <LEFT_PAREN> operand = additiveExpression(null,0) [<COMMA> format = datetimeStringFormat()] [ <COMMA> length = length() ] <RIGHT_PAREN>
+     |
+     charType = charOrVarchar()
+     <LEFT_PAREN>
+     operand = additiveExpression(null,0)
+     [
+        <COMMA> (
+            format = datetimeStringFormat()
+        |
+            length = length()
+        )
```

Comment:
Ah, now I see why the regression is there. Because they both start with `<COMMA>`, and the length one is shadowed. Thanks for the fix!

Reply:
👍